### PR TITLE
Updates in 'Build Release" GH action.

### LINKED ISSUE #305: Update action-release to run as a composite step
**Is your enhancement related to a problem? Please describe.**
It's a Bash script so it should be able to run directly on the runner, doesn't need to spin up Docker.

**Describe the solution you'd like**

**Designs**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by iamdharmesh**
### Description of the Change
This PR updates the `Build Release` workflow by removing action-release action. Currently, the `action-release` action is used to upload build files to the `stable` branch on push to the trunk. This PR updates the workflow to use the **composer archive** feature and the `s0/git-publish-subdir-action@develop` action(got the reference from [@10up/cypress-wp-utils](https://github.com/10up/cypress-wp-utils/) repo) to push build files to the branch. 

Closes #305
### Alternate Designs
Instead of using another approach to push build files to the `stable` branch, we may convert the `action-release` action steps as composite steps of the workflow file.

### Verification Process
We will need to merge this PR to the `trunk` branch to verify the Build Release action.
**Note: I have verified this PR by creating the temporary [TEST GH action](https://github.com/10up/classifai/runs/6707978701?check_suite_focus=true)(run on PR) and pushing the build files to the [`test-305`](https://github.com/10up/classifai/tree/test-305) branch.** (Will remove the `test-305` branch` one this PR get merged)

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Changelog Entry

> Changed: Updates in 'Build Release" GH action.

### Credits
Props @iamdharmesh @helen

### CODE REVIEW

**Review by dinhtungdu — COMMENTED** [review]
@iamdharmesh Thanks for working on this PR! The code looks good to me! `s0/git-publish-subdir-action` is a JavaScript action so I don't worry about the performance. Just curious why you choose it over converting the existing action to composite?

**Comment by iamdharmesh** [comment]
@dinhtungdu, not any specific major reason for choosing it, but I feel this new approach is simpler and we don't have to maintain the code(eg: [recent report](https://10up.slack.com/archives/C0WUHTSHH/p1653582333228009)) of our custom steps while we already have active action for it.

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+hookdoc-conf.json export-ignore
+LICENSE.md export-ignore
+package-lock.json export-ignore
+package.json export-ignore
+phpcs.xml.dist export-ignore
+phpunit.xml.dist export-ignore
+portkey.json export-ignore
+README.md export-ignore
+webpack.config.js export-ignore
+wpacceptance.json
```
Comment (dinhtungdu) [inline .gitattributes]: We don't use WP Acceptance anymore, should we remove this too?

**Review by dinhtungdu — DISMISSED** [review]
Left a minor comment but this is LGTM so pre-approving.

File: .gitattributes (hunk 1 above)
Comment (iamdharmesh) [reply .gitattributes]: Nice catch! Thanks for pointing this out @dinhtungdu

**Comment by jeffpaul** [comment]
@dkotter would be good to get your review on this before merging as you have the most recent info from Helen on how this used to / currently works and can best assess if this updated approach is good to go or needs further tweaks.

**Review by dkotter — APPROVED** [review]

**Pull request merged**